src/runtime/recipe/recipe-resolver.ts

```diff
@@ -123,13 +144,19 @@ export class ResolveWalker extends RecipeWalker {
   }
 
   onPotentialSlotConnection(_recipe: Recipe, particle: Particle, slotSpec: ConsumeSlotConnectionSpec) {
+    const error = (label: string) => {
+      if (this.options && this.options.errors) {
+        this.options.errors.set(particle, label);
+      }
+      return [];
+    };
     const arc = this.arc;
     const {local, remote} = SlotUtils.findAllSlotCandidates(particle, slotSpec, arc);
     const allSlots = [...local, ...remote];
 
     // SlotUtils handles a multi-slot case.
     if (allSlots.length !== 1) {
-      return undefined;
+      return error('There is not just one slot for this slot spec?');
```

Comment:
question mark at the end?

Reply:
Forgot that I wanted to improve the error message (have done so with your suggestions from above).

Reply:
Awesome, thanks.